src/Nethermind/Nethermind.BeaconNode.Storage/MemoryStore.cs

```diff
@@ -62,37 +65,115 @@ public class MemoryStore : IStore
         }
 
         public Checkpoint BestJustifiedCheckpoint { get; private set; }
-        public IReadOnlyDictionary<Hash32, BeaconBlock> Blocks { get { return _blocks; } }
-        public IReadOnlyDictionary<Hash32, BeaconState> BlockStates { get { return _blockStates; } }
-        public IReadOnlyDictionary<Checkpoint, BeaconState> CheckpointStates { get { return _checkpointStates; } }
         public Checkpoint FinalizedCheckpoint { get; private set; }
         public ulong GenesisTime { get; }
         public Checkpoint JustifiedCheckpoint { get; private set; }
-        public IReadOnlyDictionary<ValidatorIndex, LatestMessage> LatestMessages { get { return _latestMessages; } }
         public ulong Time { get; private set; }
 
-        public void SetBestJustifiedCheckpoint(Checkpoint checkpoint) => BestJustifiedCheckpoint = checkpoint;
+        public Task SetBestJustifiedCheckpointAsync(Checkpoint checkpoint)
+        {
+            BestJustifiedCheckpoint = checkpoint;
+            return Task.CompletedTask;
+        }
 
-        public void SetBlock(Hash32 signingRoot, BeaconBlock block) => _blocks[signingRoot] = block;
+        public Task SetBlockAsync(Hash32 signingRoot, BeaconBlock block)
```

Comment:
ValueTasks here?

Reply:
I'm not sure how common synchronous completion will be here, but you might have better insight from eth1. As storage tasks, the Set() is probably going to involve some write to persistent storage, but the Get() methods should be mostly reading from a cache so I guess would want to mostly be synchronous. 

I would probably change as part of optimisation later, unless you have some insights from Eth 1. (I know the current hack memory based implementation is all just synchronous, but that won't be the case with real storage).